internal/telemetry/collector/local.go

```diff
@@ -0,0 +1,560 @@
+	case gpu <= 32:
+		return prefix + "17-32"
+	case gpu <= 64:
+		return prefix + "33-64"
+	case gpu <= 128:
+		return prefix + "65-128"
+	case gpu <= 256:
+		return prefix + "129-256"
+	default:
+		return prefix + "257+"
+	}
+}
+
+var kubeVersionBuckets = map[string]struct{}{
+	"1.29": {},
+	"1.30": {},
+	"1.31": {},
+	"1.32": {},
+	"1.33": {},
+	"1.34": {},
+	"1.35": {},
+	"1.36": {},
+	"1.37": {},
+}
+
+func bucketKubeVersion(v string) string {
+	if _, ok := kubeVersionBuckets[v]; ok {
+		return "node.info.kubeVersion:" + v
+	}
+	return "node.info.kubeVersion:other"
+}
+
+func bucketOS(osname string) string {
+	prefix := "node.info.os:"
+	switch v := strings.ToLower(osname); v {
+	case "linux", "windows":
+		return prefix + v
+	default:
+		return prefix + "other"
+	}
+}
+
+func bucketArch(arch string) string {
+	prefix := "node.info.arch:"
+	switch v := strings.ToLower(arch); v {
+	case "amd64", "arm64":
+		return prefix + v
+	default:
+		return prefix + "other"
+	}
+}
+
+func bucketProvider(p string) string {
+	// (bootstrap|control-plane|infrastructure)-<provider>
+	// NOTE: very opinionated metric, let us reflect whether we actually need it
+	return "providers:" + p
+}
+
+func bucketOperatorInstalled(typ string, v bool) string {
+	return "gpu.operator_installed." + typ + ":" + strconv.FormatBool(v)
+}
+
+func bucketSyncMode(mode string) string {
+	return "syncMode:" + strings.ToLower(mode)
+}
+
+func bucketPodsWithGPURequests(n uint64) string {
+	prefix := "pods.with_gpu_requests:"
+	switch {
+	case n == 0:
+		return prefix + "0"
+	case n <= 4:
+		return prefix + "1-4"
+	case n <= 16:
+		return prefix + "5-16"
+	case n <= 64:
+		return prefix + "17-64"
+	case n <= 256:
+		return prefix + "64-256"
+	default:
+		return prefix + "257+"
+	}
+}
+
+func bucketKubevirtVMIs(n int) string {
+	prefix := "kubevirt.vmis:"
+	switch {
+	case n == 0:
+		return prefix + "0"
+	case n <= 5:
+		return prefix + "1-5"
+	case n <= 10:
+		return prefix + "6-10"
+	case n <= 20:
+		return prefix + "11-20"
+	case n <= 50:
+		return prefix + "21-50"
+	default:
+		return prefix + "51+"
+	}
+}
+
+func bucketTemplate(name string) string {
+	return "template:" + name
+}
+
+func bucketGPUGiCapacity(typ string, bytes uint64) string {
+	prefix := fmt.Sprintf("gpu.%s.gibibytes_capacity:", typ)
+	switch {
+	case bytes == 0:
+		return prefix + "0"
+	case bytes <= 4<<30:
+		return prefix + "1-4"
+	case bytes <= 8<<30:
+		return prefix + "4-8"
+	case bytes <= 16<<30:
+		return prefix + "8-16"
+	case bytes <= 32<<30:
+		return prefix + "16-32"
+	case bytes <= 64<<30:
+		return prefix + "32-64"
+	case bytes <= 128<<30:
+		return prefix + "64-128"
+	case bytes <= 256<<30:
+		return prefix + "128-256"
+	case bytes <= 512<<30:
+		return prefix + "256-512"
+	case bytes <= 1024<<30:
+		return prefix + "512-1024"
+	default:
+		return prefix + "1024+"
+	}
+}
+
+func bucketGPUGiUsed(typ string, bytes uint64) string {
+	prefix := fmt.Sprintf("gpu.%s.gibibytes:", typ)
```

Comment:
I wonder if that is expected to have reqs in GiB rather than in MiB. What would be a better option in terms of the requests? The same question is applicable for the capacity (but I reckon GiB is okay in case of the capacity)